pywbem/cim_obj.py

```diff
@@ -4799,63 +4866,78 @@ def tocimxml(self):
         :class:`~pywbem.CIMProperty` object,
         as an instance of an appropriate subclass of :term:`Element`.
 
-        The returned CIM-XML representation is consistent with :term:`DSP0201`.
+        The returned CIM-XML representation is a `PROPERTY`,
+        `PROPERTY.REFERENCE`, or `PROPERTY.ARRAY` element dependent on the
+        property type, and consistent with :term:`DSP0201`. Note that
+        array properties cannot be of reference type.
 
-        The order of qualifiers is preserved.
+        The order of qualifiers in the returned CIM-XML representation is
+        preserved from the :class:`~pywbem.CIMProperty` object.
         """
 
+        qualifiers = [q.tocimxml() for q in self.qualifiers.values()]
+
         if self.is_array:
+            assert self.type != 'reference'
```

Comment:
Isn't this a real function change? I have not yet looked but is it true that we disallow arrays of references.

Reply:
Yes, DSP0004 disallows arrays of reference properties (it does allow parameters that are arrays of references, though).